src/CMakeLists.txt

```diff
@@ -248,13 +248,73 @@ set(BOOTLOADER-SOURCES
   ${DRIVER-SOURCES}
 )
 
+get_property(SAMD51A_INCLUDES TARGET samd51a-ds PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(ASF4_MIN_INCLUDES TARGET asf4-drivers-min PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(ASF4_INCLUDES TARGET asf4-drivers PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+get_property(CMSIS_INCLUDES TARGET CMSIS PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+
+set(RUST_INCLUDES ${INCLUDES} ${SAMD51A_INCLUDES} ${ASF4_INCLUDES} ${ASF4_MIN_INCLUDES} ${CMSIS_INCLUDES})
+
+if(CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+  set(RUST_PROFILE "debug")
+else()
+  set(RUST_PROFILE "release")
+  set(RUST_CARGO_FLAG "--release")
+endif()
+
+set(LIBBITBOX02_NEW_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/rust/bitbox02-new)
+set(LIBBITBOX02_NEW_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/rust/bitbox02-new/target)
+set(LIBBITBOX02_NEW ${CMAKE_CURRENT_BINARY_DIR}/rust/bitbox02-new/target/thumbv7em-none-eabi/${RUST_PROFILE}/libbitbox02_new.a)
+
+find_program(CBINDGEN cbindgen)
+find_program(CARGO cargo)
+
+add_custom_target(rust
+  # Build rust library
+  COMMAND
+    ${CMAKE_COMMAND} -E env SYSROOT=${CMAKE_SYSROOT} INCLUDES="${RUST_INCLUDES}" ${CARGO} build --manifest-path ${LIBBITBOX02_NEW_SOURCE_DIR}/Cargo.toml --target-dir ${LIBBITBOX02_NEW_BINARY_DIR} ${RUST_CARGO_FLAG} --target thumbv7em-none-eabi
+  # Rust packages stdlib functions which we must remove/weaken
+  COMMAND
+    ${CMAKE_OBJCOPY} -N memset -N memcpy -N memmove -W memcmp ${LIBBITBOX02_NEW}
+  # Generate c-headers for the rust library
+  COMMAND
+    ${CBINDGEN} --config ${CMAKE_CURRENT_SOURCE_DIR}/rust/cbindgen.toml --output ${CMAKE_CURRENT_BINARY_DIR}/rust/bitbox02_new.h ${LIBBITBOX02_NEW_SOURCE_DIR}
+  BYPRODUCTS
+    ${LIBBITBOX02_NEW}
+    ${CMAKE_CURRENT_BINARY_DIR}/rust/bitbox02_new.h
+)
+
+# TODO(nc): Move to root cmakefile when api-docs is merged
+add_custom_target(doc-rust
+  COMMAND
+    ${CMAKE_COMMAND} -E env SYSROOT=${CMAKE_SYSROOT} INCLUDES="${RUST_INCLUDES}" ${CARGO} doc --manifest-path ${LIBBITBOX02_NEW_SOURCE_DIR}/Cargo.toml --target-dir ${CMAKE_BINARY_DIR}/doc-rust ${RUST_CARGO_FLAG} --target thumbv7em-none-eabi
+  COMMAND
+    echo "See docs at file://${CMAKE_BINARY_DIR}/doc-rust/thumbv7em-none-eabi/doc/bitbox02_new/index.html"
+)
+
+add_library(bitbox02_new STATIC IMPORTED)
```

Comment:
What does `_new` mean? Would be good to add a comment to explain what this is, and maybe pick a descriptive name.

Reply:
Basically it means "functionality implemented in rust". Should I call it `bitbox02_rust` instead?

Reply:
yes please :)